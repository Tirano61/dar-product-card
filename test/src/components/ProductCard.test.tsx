import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import ProductCard from "../../../src/components";
import { product1 } from "../data/products";

describe('ProductCard', () => {
  test('Debe  ostrar el componente correctamente', () => {
    const wrapper = render(
      <ProductCard product={product1}>
        {
          () => (
            <h1>Product Card</h1>
          )
        }
      </ProductCard>
    )

    expect(wrapper.asFragment()).toMatchSnapshot();

  });

  test('Debe incrementar el contador', async () => {
    const wrapper = render(
      <ProductCard product={product1} >
        {
          ({ count, increaseBy }) => (
            <>
              <h1>Product Card</h1>
              <span data-testid="span">{count}</span>
              <button data-testid="button" onClick={() => increaseBy(1)}>1</button>
            </>
          )
        }
      </ProductCard>
    )

    let tree = wrapper.asFragment();

    expect(tree).toMatchSnapshot();

    fireEvent.click(screen.getByRole('button'));

    expect(screen.getByTestId('span').textContent).toBe('1')

  });
});
