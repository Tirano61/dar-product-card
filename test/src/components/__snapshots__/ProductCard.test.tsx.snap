// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe  ostrar el componente correctamente 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
  </div>
</DocumentFragment>
`;

exports[`ProductCard Debe incrementar el contador 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
    <span
      data-testid="span"
    >
      0
    </span>
    <button
      data-testid="button"
    >
      +1
    </button>
  </div>
</DocumentFragment>
`;
